docs/dev/repository_layout.md

```diff
@@ -0,0 +1,216 @@
+# ScyllaDB repository layout
+
+
+This document is meant to provide a helicopter view over the ScyllaDB repository.
+
+It’d make us very happy if this document will achieve the following objectives:
+* shorten the ramp up time of an onboarding core engineer
+* improve code search locality during development of the first few tasks
+* map the theoretical knowledge one acquires during the first few days to actual source files on disk
+* get a better chance of understanding the project top-down
+
+Please note the source code is in an ever changing state with continuous effort being put in to refactor, decouple, isolate components. The source code hierarchy still looks flat with many source files located in the root directory, so some of these files might move in the future, some component dirs might get split, moved or removed altogether, we’ll do our best to keep the doc accurate.
+
+`alternator/`
+
+This is the location of the Alternator project which provides interoperability with Amazon DynamoDB, here’s a Scylla Univerisity [course](https://university.scylladb.com/courses/scylla-alternator/) on it. 
+
+`abseil/`
+
+Bundled abseil library
+
+`api/`
+
+This location contains source files managing Scylla REST APIs. ScyllaDB is being managed using the APIs defined here, the nodetool utility for instance uses such APIs to expose operations. The precursor of these APIs was Cassandra’s JMX.
+
+`auth/`
+
+Everything related to authorization and authentication between the clients and Scylla nodes.
+
+`bin/`
+
```

Comment:
Please elaborate on why do we keep binaries as part of the tree.
When are these binaries created, do we keep them in git (???), etc.

Reply:
we dont, they're just some wrapper scripts